src/app/pages/courses-page/components/courses-list/courses-list.component.ts

```diff
@@ -8,30 +7,40 @@ import Course from '../course/course.types';
   templateUrl: './courses-list.component.html',
   styleUrls: ['./courses-list.component.scss'],
 })
-export class CoursesListComponent implements DoCheck{
-  @Input() searchCourse: string;
-
-  private courses: Course[];
+export class CoursesListComponent implements OnInit, OnChanges {
+  start = 0;
+  count = 5;
+  sort = 'date';
+  @Input() textFragment = '';
+  coursesList: Array<Course> = [];
 
   constructor(
-    private filterPipe: FilterPipe,
-    private coursesService: CoursesService,
-    ) {}
+    public coursesService: CoursesService,
+  ) {
+    this.coursesService.coursesChanged.subscribe(() => this.loadData());
+  }
+
+  async onSearch(text): Promise<void> {
+    this.textFragment = text;
+    await this.loadData();
+  }
 
-  ngDoCheck(): void {
-    this.courses = this.coursesService.list;
+  async ngOnInit(): Promise<void> {
+    await this.loadData();
   }
 
-  get coursesList(): Array<Course> {
-    return this.courses;
+  ngOnChanges(changes: SimpleChanges): void {
+    const item: SimpleChange = changes.item;
```

Comment:
Please review this component. You don't need so much loadData call and if you use subscribe you need to unsubscribe.

Reply:
I added an unsubscribe from all subscriptions